src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -489,6 +482,7 @@ static uct_cm_ops_t uct_rdmacm_cm_ops = {
 static uct_iface_ops_t uct_rdmacm_cm_iface_ops = {
     .ep_pending_purge         = ucs_empty_function,
     .ep_disconnect            = uct_rdmacm_cm_ep_disconnect,
+    .cm_ep_conn_notify        = uct_rdmacm_cm_ep_conn_notify,
```

Comment:
should not `uct_rdmacm_cm_ep_conn_notify` and `ep_disconnect` be part of `uct_cm_ops_t`? 
if I remember correctly, `ep_destroy` is here due to backward compatibility reason.

Reply:
we access the ep functions from the iface (the dummy iface that was created for this purpose)
uct_rdmacm_cm_ep_conn_notify accepts a uct_ep_t as arg which only has a uct_iface_h inside

Reply:
> we access the ep functions from the iface (the dummy iface that was created for this purpose)

this makes sense for ep create/destroy/etc to provide backward compatibility.
we can cast `uct_ep_h` to `uct_cm_base_ep_t *` and access CM's ops directly from disconnect/notify and other CM only APIs. What I'm missing?

Reply:
btw, `uct_iface_ops_t` is a part of UCT/API as well, so adding extra functions to this structure we will need to maintain it

Reply:
@yosefe should we move the notify function to the cm_ops? 
would require a change in the tl.h file (under api dir) and adding uct_cm_t to the uct_cm_base_ep_t.

Reply:
no: cm_ops are global, they do not accept an object, they create new objects
when we have ep we use iface_ops